File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.cpp
class Solution {
public:
    vector<int> deckRevealedIncreasing(vector<int>& deck) {
        deque<int> dq;

        sort(deck.begin(), deck.end());
        int i = deck.size() - 1;

        dq.push_front(deck[i--]);

        while(i >= 0) {
            dq.push_front(dq.back());
            dq.pop_back();
            dq.push_front(deck[i--]);
        }

        i = 0;

        while(!dq.empty()) {
            deck[i++] = dq.front();
            dq.pop_front();
        }

        return deck;
    }
};

File: 0950-reveal-cards-in-increasing-order/0950-reveal-cards-in-increasing-order.java
class Solution {
    public int[] deckRevealedIncreasing(int[] deck) {
        Deque<Integer> dq = new ArrayDeque<>();
        
        Arrays.sort(deck);
        int i = deck.length - 1;
        
        dq.offerLast(deck[i--]);
        
        while(i >= 0){
            dq.offerFirst(dq.pollLast());
            dq.offerFirst(deck[i--]);
        }
        
        i = 0;
        
        while(!dq.isEmpty()){
            deck[i++] = dq.pollFirst();
        }
        
        return deck;
    }
}